src/Workspaces/Core/Portable/LanguageServices/DeclaredSymbolFactoryService/AbstractDeclaredSymbolInfoFactoryService.cs

```diff
@@ -110,26 +150,154 @@ protected static void Intern(StringTable stringTable, ArrayBuilder<string> build
-                : string.Concat(GenericTypeNameManglingString, arity.ToString(CultureInfo.InvariantCulture));
+            var project = document.Project;
+            var stringTable = SyntaxTreeIndex.GetStringTable(project);
+            var rootNamespace = this.GetRootNamespace(project.CompilationOptions);
+
+            using var _1 = PooledDictionary<string, string>.GetInstance(out var aliases);
+
+            foreach (var usingAlias in GetUsingAliases((TCompilationUnitSyntax)root))
+            {
+                if (this.TryGetAliasesFromUsingDirective(usingAlias, out var current))
+                    AddAliases(aliases, current);
+            }
+
+            foreach (var child in GetChildren((TCompilationUnitSyntax)root))
+                AddDeclaredSymbolInfos(root, child, stringTable, rootNamespace, declaredSymbolInfos, aliases, extensionMethodInfo, "", "", cancellationToken);
         }
 
-        public abstract bool TryGetDeclaredSymbolInfo(StringTable stringTable, SyntaxNode node, string rootNamespace, out DeclaredSymbolInfo declaredSymbolInfo);
+        private void AddDeclaredSymbolInfos(
+            SyntaxNode container,
+            TMemberDeclarationSyntax memberDeclaration,
+            StringTable stringTable,
+            string rootNamespace,
+            ArrayBuilder<DeclaredSymbolInfo> declaredSymbolInfos,
+            Dictionary<string, string> aliases,
+            Dictionary<string, ArrayBuilder<int>> extensionMethodInfo,
+            string containerDisplayName,
+            string fullyQualifiedContainerName,
+            CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
 
-        /// <summary>
-        /// Get the name of the target type of specified extension method declaration. 
-        /// The node provided must be an extension method declaration,  i.e. calling `TryGetDeclaredSymbolInfo()` 
-        /// on `node` should return a `DeclaredSymbolInfo` of kind `ExtensionMethod`. 
-        /// If the return value is null, then it means this is a "complex" method (as described at <see cref="SyntaxTreeIndex.ExtensionMethodInfo"/>).
-        /// </summary>
-        public abstract string GetReceiverTypeName(SyntaxNode node);
+            if (memberDeclaration is TNamespaceDeclarationSyntax namespaceDeclaration)
+            {
+                var innerContainerDisplayName = GetContainerDisplayName(memberDeclaration);
+                var innerFullyQualifiedContainerName = GetFullyQualifiedContainerName(memberDeclaration, rootNamespace);
 
-        public abstract bool TryGetAliasesFromUsingDirective(SyntaxNode node, out ImmutableArray<(string aliasName, string name)> aliases);
+                foreach (var usingAlias in GetUsingAliases(namespaceDeclaration))
+                {
+                    if (this.TryGetAliasesFromUsingDirective(usingAlias, out var current))
+                        AddAliases(aliases, current);
+                }
```

Comment:
this logic is incorrect (but was incorrect originally as well).  If you have multiple nested namespaces (possibly as symbols) one namespace's aliases with bleed into the others.   HOwever, this is probably very rare in practice (certainly no one has reported an issue there), so it hasn't been an issue.